src/Verilator.cpp

```diff
@@ -526,6 +526,54 @@ static void process() {
     // Note early return above when opt.cdc()
 }
 
+static void execBuildJob() {
+    if (!v3Global.opt.build()) return;
+
+    std::stringstream cmd;
+    const V3StringList& makeFlags = v3Global.opt.makeFlags();
+    const int jobs = v3Global.opt.buildJobs();
+    if (v3Global.opt.gmake()) {  // If both gmake and cmake are chosen, use gmake to build.
+        cmd <<  v3Global.opt.makeCmd();
+        cmd << " -C " << v3Global.opt.makeDir();
+        cmd << " -f " << v3Global.opt.prefix() << ".mk";
+        if (jobs == 0) {
+            cmd << " -j";
+        } else if (jobs > 1) {
+            cmd << " -j " << jobs;
+        }
+        for (V3StringList::const_iterator it = makeFlags.begin(); it != makeFlags.end(); ++it) {
+            cmd << ' ' << *it;
+        }
+    } else {
+        UASSERT(v3Global.opt.cmake(), "at least cmake or gmake must be chosen in V3Options.cpp");
+        cmd << "cd " << v3Global.opt.makeDir() << " && ";
+        cmd << "cmake";
+        for (V3StringList::const_iterator it = makeFlags.begin(); it != makeFlags.end(); ++it) {
+            cmd << ' ' << *it;
+        }
+        cmd << ' ' << V3Os::getcwd() << " && ";
+        cmd << "cmake --build . ";
+        if (jobs == 0) {
+            cmd << " -j";
+        } else if (jobs > 1) {
+            cmd << " -j " << jobs;
+        }
+    }
+    const std::string cmdStr = cmd.str();
+
+    const int ret = V3Os::system(cmdStr);
+    if (ret == -1) {
+        v3fatal("Failed to execute " << cmdStr);
+    } else {
+        UASSERT(WIFEXITED(ret), "system(" << cmdStr << ") returned unexpected value of " << ret);
+        const int exit_code = WEXITSTATUS(ret);
+        UINFO(1, cmdStr << " returned exit code of " << exit_code << std::endl);
+        if (exit_code != 0) {
+            v3error(cmdStr << " exitted with " << exit_code << std::endl);
+        }
```

Comment:
Don't need if exit_code, already checked above.

suggest after this error we should exit with the code the child exited with, so upper scripts (if any) get the proper indication.

Reply:
V3Os::system() directly returns the value form system(3), so the returned value is _not_ an exit code of the child process.
http://man7.org/linux/man-pages/man3/system.3.html#RETURN_VALUE

I think the current convention is not good because V3Os::system() reveals the detailed behavior of system(3).
I will change V3Os::system() to return exit code of the child process.

exit() with the exit code sounds  good. I will do in the way.

Reply:
Exit code propagates to the caller script now.
https://github.com/verilator/verilator/blob/38af6bbae435060505b40f6cc94320698c14809d/src/Verilator.cpp#L619-L623